### PR TITLE
Fr/adding contributing file

### LINKED ISSUE #24: Add CONTRIBUTING file
Add a CONTRIBUTING file to the repo containing guidelines on how to contribute to the package. For example, package structure, coding conventions, how to test he package.

See guidance and examples in https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors

### PR DESCRIPTION
**Pull request by rodriguesfred**
### Summary
Adding CNTRIBUTING file
Changes made to code by pre-commit hook

### Related Issues

- Fixes #
- Closes #24 
- Relates to #24 

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔧 Refactoring (no functional changes, no api changes)
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test addition or improvement
- [ ] 🔨 Build/CI changes
- [ ] 🎨 Style/formatting changes

## Changes Made
- Adding CONTRIBUTING file
- Changes made to code by pre-commit hook

## Testing
None

### Test Coverage
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Manual testing performed
- [x] No tests needed (explain why below)

### Testing Details

**Test Cases:**
1. 
2. 
3. 

**Manual Testing Results:**

## Breaking Changes

- [ ] No breaking changes
- [ ] Breaking changes (describe below)

**Breaking Change Details:**

## Documentation
- [x] Documentation updated
- [x] README updated
- [ ] No documentation changes needed

**Documentation Changes:**
Moved relevant sections of README to CONTRIBUTING file
Updated CONTRIBUTING with elements from [AF guidance](https://best-practice-and-impact.github.io/qa-of-code-guidance/project_documentation.html)

## Dependencies
- [x] No new dependencies
- [ ] New dependencies added (list below)
- [ ] Dependencies updated (list below)
- [ ] Dependencies removed (list below)

**Dependency Changes:**

### Code Quality
- [ ] Code follows the project's [style guidelines](https://3.basecamp.com/4322319/buckets/25844562/documents/8777425747)
- [ ] Self-review of code completed
- [ ] Code is properly commented
- [ ] No unnecessary console.log/print statements
- [ ] No commented-out code (unless intentionally documented)

### Testing & Quality Assurance
- [ ] Tests pass locally
- [ ] Tests added for new functionality
- [ ] Edge cases considered and tested
- [ ] Error handling implemented where appropriate

### CODE REVIEW

**Comment by rodriguesfred** [comment]
@TomFarley @nickhenden updated the PR following changes needed to pass checks

**Review by TomFarley — COMMENTED** [review]
Looks good. I've added a few suggestions/comments.

File: pyproject.toml [hunk 1]
```diff
@@ -28,9 +28,10 @@ dependencies = [ # install this package in developer mode with e.g. $ uv pip ins
     "matplotlib>=3.10.3",
     "plotly>=6.1.2",
     "seaborn>=0.13.2",
+    "typeguard>=4.4.4",
 ]
 
-[dependency-groups]
+[project.optional-dependencies]
```
[the remark this answers is not in the document]
Comment (rodriguesfred) [reply pyproject.toml:34]: this did not work when the github actions ran. I think it might be because github uses pip.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+## Code style
+
+- Use meaningful variable and function names.
+- We name variables using few nouns in lowercase, e.g. `mapping_names`
+or `increment`.
+- We name functions using verbs in lowercase, e.g. `map_variables_to_names` or
+`change_values`.
+- Follow PEP 8 style guidelines for Python code. <mark>(to review)<mark>
+- Write clear, concise, and well-documented code. Include comments where necessary to explain complex logic.
+- We use the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html)
```
[the remark this answers is not in the document]
Comment (rodriguesfred) [reply CONTRIBUTING.md]: Added a line with the link to the Python style guide.
On a side note when looking at the link it mentions something about a 'ruff.toml' configuration file which I don't think we're using. Is this required?

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (TomFarley) [reply pyproject.toml:34]: Dependency groups are established in [PEP 735](https://peps.python.org/pep-0735/#example-dependency-groups-table), so pip should be fine with them, but let's stick with whatever is working.

**Review by TomFarley — APPROVED** [review]
Everything is looking good to me

**Comment by rodriguesfred** [comment]
@TomFarley @nickhenden is there anything outstanding?

**Review by nickhenden — APPROVED** [review]
Looks good :) Approved.
